### PR TITLE
NAS-114289 / 22.12 / Update serial setup in Installer's grub.cfg

### PR DESCRIPTION
**Pull request by usaleem-ix**
This commit fixes SCALE boot issue in BIOS mode on X series hardware.
In BIOS mode, we were not able to see any output from GRUB on IPMI
since serial setup was not being done properly in SCALE installer's
grub.cfg.

Moreover, currently both boot options provide serial console options
for SCALE, due to which we always see Linux console on serial since
serial settings are saved by installer. User should only select
(115200 baud) boot option if they want serial console, otherwise serial
console should not be provided.

Signed-off-by: Umer Saleem <usaleem@ixsystems.com>

### CODE REVIEW

**Comment by freqlabs** [comment]
Why the sleep?

**Comment by usaleem-ix** [comment]
> Why the sleep?

With serial input enabled, sleep helps consume buffered input. `grub.cfg` generated by `grub-install` also includes the sleep for this purpose.

**Review by freqlabs — APPROVED** [review]

File: scripts/grub.cfg [hunk 1]
```diff
@@ -16,18 +16,20 @@ if loadfont $prefix/fonts/unicode.pf2 ; then
...
   set lang=en_US
   set gfxpayload=keep
 fi
-terminal_output gfxterm
 
-serial --unit=0 --speed=9600 --word=8 --parity=no --stop=1
+serial --unit=0 --speed=115200 --word=8 --parity=no --stop=1
+terminal_output console serial
+terminal_input console serial
+sleep -i 1
```
Comment (william-gr) [inline scripts/grub.cfg:28]: What do you think of adding that comment here, the one you replied to Ryan, please?

Sounds useful enough so the next person doesn't get lost :D

Comment (usaleem-ix) [reply scripts/grub.cfg:30]: Sure, I have added the comment.

Comment (william-gr) [reply scripts/grub.cfg:30]: Thank you!

**Pull request merged**